mage/query-modules/python/gnn-link-prediction.md

```diff
@@ -0,0 +1,375 @@
+
+
+#### **Input**:
+
+| Name            | Type  | Default     | Description |
+| --------------- | ----- | ----------- | ----------- |
+| `hidden_features_size` | mgp.List[int] | `[16, 16]` | Defines the size of each hidden layer in the architecture. Input feature size is determined automatically while converting the original graph to the DGL-compatible one. |
+| `layer_type`    | str   | `graph_attn` | Supported values are `graph_sage` and `graph_attn`. |
+| `num_epochs`    | int   | `100`  | number of epochs for model training.   |
+| `optimizer`     | str   | `ADAM` |  Supported values are `ADAM` and `SGD`. |
+| `learning_rate` | float | `0.01` | optimizer's learning rate.             |
+| `split_ratio`   | float | `0.8`  | split ratio between training and validation set. There is no test dataset because it is assumed that the user first needs to create new edges in the original dataset to test a model on them. |
+| `node_features_property` | str | `features` | property name where the node features are saved. |
+| `device_type` | str | `cpu` | If the model will be trained using CPU or Cuda GPU. To be run on `cuda`, the user must set this flag to `cuda` and the system must support `cuda` execution. System’s support is checked with `torch.cuda.is_available()`. Supported values are `cpu` and `cuda`. |
+| `console_log_freq` | int | `5` | How often results will be printed. This also directly specifies which results will be returned as training and validation results when calling the training method. |
+| `checkpoint_freq` | int | `5` | Select the number of epochs on which the model will be saved. The model is persisted on disc. |
+| `aggregator` | str | `mean` | Aggregator used in GraphSAGE model. Supported values are lstm, pool, mean and gcn. |
+| `metrics` | mgp.List[str] | `[loss, accuracy, auc_score, precision, recall, f1, true_positives, true_negatives, false_positives, false_negatives]` | Metrics used to evaluate the model in training on the validation set. Additionally, epoch information will always be displayed. |
+| `predictor_type` | str | `dot` | Type of the predictor. A predictor is used for combining node scores to edge scores. Supported values are `dot` and `mlp`. |
+| `attn_num_heads` | List[int] | `[4, 1]` | `GAT` can support the usage of more than one head in each layer except the last one. The size of the list must be the same as the number of layers specified by the `hidden_features_size` parameter. |
+| `tr_acc_patience` | int | `8` | Training patience, specifies how many epochs will accuracy drop on the validation set be tolerated before the training will be stopped. |
```

Comment:
I can't comprehend this description. Are you sure it's 100% correctly written? 
If it is, can you write it to me in croatian?

Reply:
@as51340 - i can't figure out the subject - verb - object structure in this sentence... it's like there are too many verbs or something